pkg/sfu/prober.go

```diff
@@ -280,8 +331,15 @@ func (p *Prober) run() {
 			}
 
 			p.popFrontCluster(cluster)
-			continue
 		}
+
+		// determine how long to sleep
+		cluster := p.getFrontCluster()
+		if cluster == nil {
+			return
+		}
+
+		ticker.Reset(cluster.GetSleepDuration())
```

Comment:
if you're resetting this on every loop it might as well be a `time.Timer` then because the only way to exit the function is the return after the ticker fired you don't need the `defer ticker.Stop()`

Reply:
Will change that.